pandas/core/arrays/datetimes.py

```diff
@@ -1584,7 +1584,7 @@ def isocalendar(self) -> DataFrame:
     2017-01-06    4
     2017-01-07    5
     2017-01-08    6
-    Freq: D, dtype: int64
+    Freq: D, dtype: int32
```

Comment:
could split off the datetimelike bits?

Reply:
No, this change is a consequence of allowing 32bit indexes, which wasn't possible before this PR.